pandas/tests/indexes/test_base.py

```diff
@@ -678,61 +678,53 @@ def test_intersection(self):
         inter = first.intersection(first)
         assert inter is first
 
+    @pytest.mark.parametrize("idx2,keeps_nm", [
+        (Index([3, 4, 5, 6, 7], name="idx"), True),  # preserve same name
+        (Index([3, 4, 5, 6, 7], name="other"), False),  # drop diff names
+        (Index([3, 4, 5, 6, 7]), False)])
+    def test_intersection_name_preservation(self, idx2, keeps_nm):
```

Comment:
side note, we prob need to split out test_base into multiple files its growing pretty big (do this as a separate step), these might be things like: test_indexing.py, test_missing.py, test_set_ops.py etc, in essence mirroring the various sub-files we do for series ops

Reply:
Yep - I have this as the last TODO in the referenced issue so we don't lose sight of it